consensus/src/sync/light/sync_stream.rs

```diff
@@ -245,12 +244,11 @@ impl<TNetwork: Network> LightMacroSync<TNetwork> {
                             if epoch_ids.checkpoint_epoch_number() == 0 {
                                 return 0;
                             }
-                            // FIXME: Ban peer if it sends an invalid epoch_number instead of panicking.
                             Policy::first_block_of(epoch_ids.checkpoint_epoch_number() as u32)
-                                .expect("The supplied epoch number is out of bounds")
+                                .unwrap_or(u32::MAX)
                         })
-                        + Policy::blocks_per_batch()
-                        - 1;
+                        .saturating_add(Policy::blocks_per_batch())
+                        .saturating_sub(1);
```

Comment:
As blocks per batch is a positive non zero number the 1 could be moved into the saturating add, but it does not really matter.

Also looks like all of this would also be `macro_block_after()` and if so I would prefer to use it instead.